app/code/Magento/Catalog/Model/Product/Compare/ListCompare.php

```diff
@@ -67,13 +75,16 @@ public function __construct(
         \Magento\Catalog\Model\ResourceModel\Product\Compare\Item $catalogProductCompareItem,
         \Magento\Customer\Model\Session $customerSession,
         \Magento\Customer\Model\Visitor $customerVisitor,
+        \Magento\Catalog\Model\ProductFactory $catalogFactory = null,
```

Comment:
Please don't change the order of constructor parameters.
Add the new parameter after `$data` for backward compatibility.

Reply:
fixed.